com.unity.render-pipelines.universal/Runtime/DeferredLights.cs

```diff
@@ -454,18 +499,20 @@ internal void SetupLights(ScriptableRenderContext context, ref RenderingData ren
 
             DeferredShaderData.instance.ResetBuffers();
 
-            this.RenderWidth = renderingData.cameraData.cameraTargetDescriptor.width;
-            this.RenderHeight = renderingData.cameraData.cameraTargetDescriptor.height;
+            Camera camera = renderingData.cameraData.camera;
+            // Support for dynamic resolution.
+            this.RenderWidth = camera.allowDynamicResolution ? Mathf.CeilToInt(ScalableBufferManager.widthScaleFactor * camera.pixelWidth) : camera.pixelWidth;
```

Comment:
Should we apply the render scale from the pipeline asset here? I think the targetDescriptor.width differs from camera.pixelWidth due to that.

Reply:
It doesn't work for scene 127_DynamicScaling_DOF. This scene sets the dynamic resolution directly via `ScalableBufferManager` (lowest C# interface for dynamic resolution), so bypasses `UniversalRenderPipelineAsset.renderScale`.
`cameraData.renderScale` and `targetDescriptor.width` derive their values from `UniversalRenderPipelineAsset.renderScale`.